core/trino-main/src/main/java/io/trino/execution/QueryStateMachine.java

```diff
@@ -618,7 +618,7 @@ private QueryStats getQueryStats(Optional<StageInfo> rootStage, List<StageInfo>
 
         boolean isScheduled = rootStage.isPresent() && allStages.stream()
                 .map(StageInfo::getState)
-                .allMatch(state -> state == StageState.RUNNING || state == StageState.PENDING || state.isDone());
+                .anyMatch(state -> state == StageState.RUNNING || state == StageState.PENDING || state.isDone());
```

Comment:
how does it impact progress bar rendering for non-fault tolerant execution? Does it impact anything else than progress bar rendering?

Reply:
I checked all text occurrences of `isScheduled`, seems it's only used in progress bar rendering

Reply:
I would recommend to refrain from changing the `isScheduled` semantics. If it's only used for progress bar rendering can we implement a check that we need there? (or maybe add a separate boolean filed, `isAnyStageScheduled`)?

Reply:
I would argue that `anyMatch` is the right semantics here. As long as any of the stage is scheduled, then the whole query should be seen as scheduled, right?

Reply:
For a single stage the `SCHEDULED` status indicates that all splits have been scheduled. That's why I'm a little worried that it might be not intuitive for `isScheduled` to return `true` as soon as only a single stage has all it's split scheduled.

Reply:
Then for fault tolerant execution, it's not scheduled until the last stage executes, which doesn't make sense. Plus, `isScheduled` is only used in progress bar, so if we add a `isAnyStageScheduled`, then will leave `isScheduled` effectively useless (and therefore should be removed).

Reply:
Do you think we can simply rename it?